tests/test_web_sendfile_functional.py

```diff
@@ -240,24 +257,26 @@ async def handler(request):
 
     resp = await client.get("/")
     assert resp.status == 200
-    body = await resp.read()
-    assert b"hello aiohttp\n" == body
-    assert resp.headers["Content-Type"] == "application/pdf"
     assert resp.headers.get("Content-Encoding") == "gzip"
+    assert resp.headers["Content-Type"] == "application/pdf"
+    assert await resp.read() == HELLO_AIOHTTP
     resp.close()
     await resp.release()
     await client.close()
 
 
-async def test_static_file_with_gziped_counter_part_enable_compression(
-    aiohttp_client: Any, sender: Any
+@pytest.mark.parametrize("forced_compression", [None, web.ContentCoding.gzip])
+async def test_static_file_with_encoding_and_enable_compression(
+    hello_txt: pathlib.Path,
+    aiohttp_client: Any,
+    sender: Any,
+    forced_compression: Optional[web.ContentCoding],
```

Comment:
`Optional` is considered confusing, so it'd be nice not to introduce it in favor of being explicit:
```suggestion
    forced_compression: "web.ContentCoding | None",
```

Reply:
We shouldn't use the new syntax, it may cause a SyntaxError if inspected at runtime on Python 3.8/9. I'd just leave these alone and let pyupgrade handle them later. FYI, as far as I'm aware the recommendation was always to use `Optional` until it became deprecated. Now we're just waiting till we can use the new syntax.

Reply:
I see your point. Though, I think we should rely on what's tested, not on something some people remember to do occasionally. But fine, I won't insist.l

Reply:
I'm in favor of `from __future__ import annotations` as much as possible as the new syntax is so much nicer and more readable. Thats probably for another PR though.